### PR TITLE
Add timestamp field support

### PR DESCRIPTION
**Pull request by rstuven**
According to exposition format specification https://prometheus.io/docs/instrumenting/exposition_formats/

### CODE REVIEW

**Comment by rstuven** [comment]
TODO: Update docs

File: lib/counter.js [hunk 1]
```diff
@@ -47,15 +48,16 @@ function Counter(name, help, labels) {
...
  * @param {Number} value - Value to increment, if omitted increment with 1
+ * @param {(Number|Date)} timestamp - Timestamp to set the counter to
  * @returns {void}
  */
-Counter.prototype.inc = function(labels, value) {
+Counter.prototype.inc = function(labels, value, timestamp) {
 	if(isNumber(labels) || !labels) {
-		return inc.call(this, null)(labels);
+		return inc.call(this, null)(labels, value);
 	}
```
Comment (siimon) [inline lib/counter.js]: Not sure why value is added here

File: lib/gauge.js [hunk 2]
```diff
@@ -49,34 +50,37 @@ function Gauge(name, help, labels) {
...
  * @param {object} labels - Object with labels and their values
  * @param {Number} value - Value to set the gauge to, must be positive
+ * @param {(Number|Date)} timestamp - Timestamp to set the gauge to
  * @returns {void}
  */
-Gauge.prototype.set = function(labels, value) {
+Gauge.prototype.set = function(labels, value, timestamp) {
 	if(isNumber(labels)) {
-		return set.call(this, null)(labels);
+		return set.call(this, null)(labels, value);
```
Comment (siimon) [inline lib/gauge.js]: Same here, I'm probably missing something :)

File: lib/counter.js (hunk 1 above)
Comment (rstuven) [inline lib/counter.js]: If labels is a number, then the next argument (value) actually is a timestamp.

File: lib/gauge.js (hunk 2 above)
Comment (rstuven) [inline lib/gauge.js]: Same as above :)

**Comment by rstuven** [comment]
> Have you verified this so Prometheus is able to scrape metrics with time stamps?

Yes, already verified.

File: lib/counter.js (hunk 1 above)
Comment (siimon) [inline lib/counter.js]: Gotcha!

Might rename the variable then so it makes more sense, I can do that later on if you don't want to do it..

**Comment by siimon** [comment]
Another thing, we need to update the typescript definitions as well, as in the other comment, if you don't want to do it I can do it after this PR is merged

**Comment by SimenB** [comment]
Any news here? 😄

**Pull request merged**